src/chat/flagsReducers.js

```diff
@@ -39,10 +40,17 @@ const addFlagsForMessages = (state: FlagsState, messages, flags: string[]): Flag
     });
   });
 
-  return {
+  const changedState = {
     ...state,
     ...newState,
   };
+
+  // TODO: improve and add tests
```

Comment:
Consider checking newState for being an empty object `{}` (it is a bit tricky, I think lodash might have a function).

Doing a deep comparison on a large object like this is very likely not a good idea.